.github/workflows/patch-release-build.yml

```diff
@@ -23,39 +23,155 @@ jobs:
           echo "::set-output name=release-branch-name::$(echo '${{ github.event.inputs.version }}' | sed -E 's/([^.]+)\.([^.]+)\.([^.]+)/v\1.\2.x/')"
           # Sets the release-tag-name output to the version number with the last non-period element replace with a '0' and prepended with v
           echo "::set-output name=release-tag-name::$(echo '${{ github.event.inputs.version }}' | sed -E 's/([^.]+)\.([^.]+)\.([^.]+)/v\1.\2.0/')"
+
       - id: checkout-release-branch
         name: Check out release branch
         continue-on-error: true
         uses: actions/checkout@v2.3.4
         with:
           ref: ${{ steps.parse-release-branch.outputs.release-branch-name }}
+          fetch-depth: 0
+
       - id: checkout-release-tag
         name: Check out release tag
         if: ${{ steps.checkout-release-branch.outcome == 'failure' }}
         uses: actions/checkout@v2.3.4
         with:
           ref: ${{ steps.parse-release-branch.outputs.release-tag-name }}
+          fetch-depth: 0
+
       - name: Create release branch
         if: ${{ steps.checkout-release-tag.outcome == 'success' }}
         run: |
           git checkout -b ${{ steps.parse-release-branch.outputs.release-branch-name }}
           git push --set-upstream origin ${{ steps.parse-release-branch.outputs.release-branch-name }}
-  build:
+
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [ 8, 11, 15 ]
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          ref: ${{ needs.prepare-release-branch.outputs.release-branch-name }}
```

Comment:
If you reference `needs` here, shouldn't this job actually have a `needs` section?